src/main/java/com/bobocode/hoverla/bring/context/BeanInitializer.java

```diff
@@ -3,40 +3,64 @@
 import com.bobocode.hoverla.bring.annotation.Bean;
 import com.google.common.collect.Table;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
 
 import java.util.Collection;
 
 /**
  * Class, responsible for triggering bean initialization.
  *
  * <p>Its main aim is to find all dependencies of a particular {@link BeanDefinition} and pass them
- * to its {@link BeanDefinition#instance(BeanDefinition...)} method.</p>
+ * to its {@link BeanDefinition#instantiate(BeanDefinition...)} method.</p>
  *
  * @see Bean @Bean
  * @see BeanDefinition
  */
 @Slf4j
 public class BeanInitializer {
 
-
     /**
      * Triggers instantiation of all beans by passing their required dependencies.
+     *
      * @param beanDefinitionsTable {@link Table} with all {@link BeanDefinition} objects handled by current context.
      */
     public void initialize(Table<String, Class<?>, BeanDefinition> beanDefinitionsTable) {
         log.debug("Bean initialization started");
         Collection<BeanDefinition> beanDefinitions = beanDefinitionsTable.values();
+        beanDefinitions.forEach(beanDefinition -> doInitialize(beanDefinition, beanDefinitionsTable));
+    }
+
+    private void doInitialize(BeanDefinition definitionToInitialize,
+                              Table<String, Class<?>, BeanDefinition> beanDefinitionsTable) {
+        if (definitionToInitialize.isInstantiated()) {
+            return;
+        }
+        String beanName = definitionToInitialize.name();
+        log.trace("Resolving dependencies for bean definition: {}", beanName);
 
-        for (BeanDefinition beanDefinition : beanDefinitions) {
-            log.trace("Resolving dependencies for bean definition: {} - {}", beanDefinition.name(), beanDefinition.type());
-            BeanDefinition[] beanDependencies = beanDefinition.dependencies()
-                    .entrySet()
-                    .stream()
-                    .map(keyPair -> beanDefinitionsTable.get(keyPair.getKey(), keyPair.getValue()))
-                    .toArray(BeanDefinition[]::new);
+        BeanDefinition[] beanDependencies = getBeanDependencies(definitionToInitialize, beanDefinitionsTable);
+        int numberOfDependencies = ArrayUtils.getLength(beanDependencies);
```

Comment:
Why not to use just beanDependencies.length, what it gives us? :)

Reply:
Nothing actually) Perhaps a lil bit better in terms of readability...  No hurt for performance) I can change it to usual `.length` if you wish